File: frontend/src/types/actions/query/confirm.ts
import { z } from "zod";
import { integrationEnum } from "@/types/integration";
import { messageSchema } from "@/types/actions/query/base";

export const confirmRequestSchema = z.object({
  chat_history: z.array(messageSchema),
  api_key: z.string(),
  enable_verification: z.boolean(),
  integrations: z.array(integrationEnum),
  function_to_verify: z.string(),
  instance: z.string().nullable(),
});

export type ConfirmRequest = z.infer<typeof confirmRequestSchema>;

export const userVerificationSchema = z.enum(["YES", "NO"]);

File: backend/app/models/query/confirm.py
from typing import Optional

from pydantic import BaseModel

from app.models.integrations.base import Integration
from app.models.query.base import Message


class ConfirmRequest(BaseModel):
    chat_history: list[Message]
    api_key: str
    enable_verification: bool
    integrations: list[Integration]
    function_to_verify: str
    instance: Optional[str] = None
